processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -420,30 +409,54 @@ public RowBasedKey fromByteBuffer(ByteBuffer buffer, int position)
       }
```

Comment:
Maybe extract the above 9 lines into a method, for making `comparator()` body more readable?